site/app/controllers/NotificationController.php

```diff
@@ -88,6 +88,14 @@ public function viewNotificationSettings() {
     public function changeSettings() {
         //Change settings for the current user...
         $new_settings = $_POST;
+        if ($this->core->getConfig()->isEmailEnabled()) {
+            foreach ($new_settings as $key => $value) {
+                if (!substr_compare($key,'_email',strlen($key)-6)) {
+                    unset($new_settings[$key]);
+                }
+            }
+        }
```

Comment:
This doesn't address what I was talking about (and since the input fields aren't even shown on the form, has no effect).

It's rather the next `if` block, namely the:
```
            $values_not_sent = array_diff(self::NOTIFICATION_SELECTIONS, array_keys($new_settings));
            foreach(array_values($values_not_sent) as $value) {
                $new_settings[$value] = 'false';
            }
```

where any setting that wasn't part of the `$_POST` payload is just implicitly set to false, which would include the email settings, regardless of their previous setting. Not sure why this was added, but can probably remove it.

Reply:
I split the constant array NOTIFICATION_SELECTIONS into one for notification and one for email and decided at run time to add EMAIL_SELECTIONS to the property $selections. That way the changeSettings() function can ignore the email settings. I tried just deleting making the unset settings to false but if all check boxes are false the update part of the query would throw a syntax error as there would be no keys to update.